src/main/java/duke/ui/UI.java

```diff
@@ -0,0 +1,115 @@
+package duke.ui;
+
+import java.util.Scanner;
+
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+
+/**
+ * Class represents UI functionalities.
+ */
+public class UI {
+    private Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Displays greeting sentence.
+     */
+    public void onEnter() {
+        System.out.println("Hello! I'm Plaudern\nWhat can I do for you?");
+    }
+
+    /**
+     * Displays task list.
+     */
+    public String showList(TaskList taskList) {
+        return "Here are the tasks in your list:\n" + taskList.listTask();
+    }
+
+    /**
+     * Displays matching task list.
+     *
+     * @param taskList List to be displayed.
+     */
+    public String showMatchedList(TaskList taskList) {
+        return "Here are the matching tasks in your list:\n" + taskList.listTask();
+    }
+
+    /**
+     * Displays success message after add task to the list.
+     *
+     * @param task  Task added.
+     * @param count Number of tasks in the list after the addition.
+     */
+    public String onAddSuccess(Task task, Integer count) {
```

Comment:
the "success" word may not be necessary since there is already exception handling in your code that produces the error messages. hence, there is no onAddFailure() function to print the error message. I think the function name can also be more descriptive. e.g showAddedTask()